05.wc/wc.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  stdin = $stdin
+  options, filenames = parse_options(ARGV)
+  if filenames.any?
+    files = convert_to_count_contents(read_files(filenames))
+    length = find_max_length(files)
+    show_count_contents(files, options, length)
+  elsif !stdin.eof?
+    stdin_files = []
+    stdin_files << {
+      name: nil,
+      content: stdin.readlines
+    }
+    inputs = convert_to_count_contents(stdin_files)
+    length = 0
+    show_count_contents(inputs, options, length)
+  end
+end
+
+def parse_options(argv)
+  option = OptionParser.new
+  options = {}
+  option.on('-l') { options[:line] = true }
+  option.on('-w') { options[:word] = true }
+  option.on('-c') { options[:character] = true }
+
+  filenames = option.parse(argv)
+  %i[line word character].each { |key| options[key] = true } if options.empty?
+  [options, filenames]
+end
+
+def read_files(files)
+  files.map do |file|
+    File.open(file) do |f|
+      {
+        name: file,
+        content: f.readlines
+      }
+    end
+  end
+end
+
+def convert_to_count_contents(files)
+  files.map do |file|
+    {
+      filename: file[:name],
+      line: file[:content].join.count("\n").to_s,
+      word: file[:content].map { |test| test.split.size }.sum.to_s,
+      character: file[:content].map(&:bytesize).sum.to_s
+    }
+  end
+end
+
+def find_max_length(files)
+  total = [
+    files.map { |file| file[:line].to_i }.sum.to_s,
+    files.map { |file| file[:word].to_i }.sum.to_s,
+    files.map { |file| file[:character].to_i }.sum.to_s
+  ]
+  total.map(&:length).max
+end
+
+def show_count_contents(contents, options, max_length)
+  contents.each do |content|
+    %i[line word character].each do |key|
+      next if !options[key]
+
+      if !content[:filename] && options.size != 1
+        print "#{content[key].rjust(7, ' ')} "
+      else
+        print "#{content[key].rjust(max_length, ' ')} "
+      end
```

Comment:
ここも `if` で書くのは変化するところだけにできるとよいですね

```suggestion
      rjust_length = content[:filename] || options.size == 1 ? max_length : 7
      print "#{content[key].rjust(rjust_length, ' ')} "
```